src/eckit/maths/Qhull.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+
+#pragma once
+
+
+#include <string>
+#include <vector>
+
+#include "eckit/exception/Exceptions.h"
+#include "eckit/maths/ConvexHull.h"
+
+#include "libqhullcpp/Qhull.h"
+
+
+namespace eckit::maths {
+
+
+class Qhull {
+public:
+    // -- Types
+
+    using coord_t = std::vector<double>;
+
+    struct Exception : eckit::Exception {
+        Exception(const std::string& what, int _errorCode, const std::string& _command) :
+            eckit::Exception(what), errorCode(_errorCode), command(_command) {}
+        const int errorCode;
+        const std::string& command;
+    };
+
+    // -- Constructors
+
+    Qhull(size_t N, const coord_t& coord, const std::string& command);
+
+    // -- Overridden methods
+
+    std::vector<size_t> list_vertices() const;
+    std::vector<std::vector<size_t>> list_facets() const;
+    std::vector<Triangle> list_triangles() const;
```

Comment:
list_triangles() only works in particular case I suppose. This should be documented.

Reply:
Yes, but triangulation is rather the common usage in most cases of both convex hull and delaunay triangulations, that's why I've included it. The Triangle element is a "flat" as can be. I didn't go as far as to specify a Polygon element though, I think it's overkill (and there is intersection with the ::geo/::geometry parts too), but it would be correct though.

In random lists of points (in most geometry cases for our applications, but not necessarily the case for polytope (where many of the axes are aligned or of the same size), the result is a triangulation. The common examples using Qhull (that I found) are with "Qt" which enforces this, so,  I think it is natural to have it as base functionality